### DISCUSSION TITLE
BarChart conditional colors

### DISCUSSION THREAD

**Discussion by mateuszurbanek** [plouc/nivo#2115] [Q&A]
Hello,

I'm struggling with a problem and I don't know anymore what to do. Is there any way that I can make BarChart with custom colors which are rendering conditionally? For example I have a BarChart with data where the color of the Bar defines it's value i.e.: values in range from 1 - 1.49 will have red Bar, values in range from 1.5 to 3.0 will be yellow and greater than 3 should be green. Is there any way I can achieve that?

**Comment by plouc** [comment]
@illumiMatii, yes, you can pass a function to `colors`, as it's done for example here: https://nivo.rocks/storybook/?path=/story/bar--using-custom-color.

**Reply by Ayush11139** [reply]
@plouc Please tell me how to place bar labels above bars in bar graph. If possible provide a working code. Thanks
